torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2079,6 +2142,9 @@ void ProcessGroupNCCL::watchdogHandler() {
             ", last completed work: ",
             pgStatus_->lastCompletedSeq);
 
+        // broadcast remote error signal to all other ranks in this specific PG.
+        broadcastSignal(store_, std::string(REMOTE_ERROR_SIGNAL), rank_);
+
```

Comment:
I haven't checked whether `broadcastSignal` could be blocking or not. Let's be careful when putting potentially blocking call in watchdog.

Reply:
The same functionality was in watchdog thread in the original code